### ISSUE TITLE
There is a problem with the read and write strategy (selecting nodes)

### ISSUE THREAD

**Issue by beck-8** [minio/minio#19775]
## NOTE
If this case is urgent, please subscribe to [Subnet](https://min.io/pricing) so that our 24/7 support team may help you faster.

## Expected Behavior

Data on this disk should not be read until the disk damage has been repaired.
## Current Behavior

Before disk damage is repaired, it appears that data is being read from the damaged disk. This results in slow/stuck reading of data and the HeadObject taking a very long time to respond. If the machine I'm repairing the disk on is completely shut down at this point, normal read speeds will resume.
## Possible Solution

- confirm question
- Modify strategy
## Steps to Reproduce (for bugs)

My environment is: 4*4（4 storage machines, 4 * 14TiB hard drives）

1. Write data to more than 60% of the disk capacity
2. Replace a data disk to simulate a failure(Make it in the process of failure recovery)
3. Read the data on this disk (you may want to observe in advance what data is written on this disk)
4. Observed slowness

## Context

The disk failure and damage replacement process should be imperceptible; performance should not be affected.
## Your Environment

* Version used (`minio --version`):
```bash
root@node1:~# minio -v
minio version RELEASE.2024-03-10T02-53-48Z (commit-id=88a89213ff6663f2d3ee1de7f3293497ab9844e8)
Runtime: go1.21.8 linux/amd64
License: GNU AGPLv3 <https://www.gnu.org/licenses/agpl-3.0.html>
Copyright: 2015-2024 MinIO, Inc.
```
* Server setup and configuration:
```bash
minio server --console-address :9001 --address :9000 http://node{1...4}.net:9000/mnt/disk{1...4} 
```
* Operating System and version (`uname -a`):
```bash
root@node1:~# uname -a
Linux node1 5.15.0-91-generic #101-Ubuntu SMP Tue Nov 14 13:30:08 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by beck-8** [comment]
https://github.com/minio/minio/issues/18716

**Comment by harshavardhana** [comment]
There is no real issue reporting here, please upgrade your setup to the new release and let us know.

**Comment by ravindk89** [comment]
@beck-8 two things:

1) do you see the same results with a newer release of MinIO? We have to compare based on the latest code, even if it is only one or two months back. 

2) Can you run `mc admin trace -a --funcname "os.*" --funcname "storage.*"` during a read operation? 

We need more data here , specifically to drive the claim

>  it appears that data is being read from the damaged disk

**Comment by beck-8** [comment]
The latest version has the same result.

**Comment by harshavardhana** [comment]
> Before disk damage is repaired, it appears that data is being read from the damaged disk. This results in slow/stuck reading of data and the HeadObject taking a very long time to respond. If the machine I'm repairing the disk on is completely shut down at this point, normal read speeds will resume.

A healing disk participates in the overall process. If your drive has low IOPs, you should tune your healing speed to ensure that it has not lost all its IOPs.

We cannot just go ahead and ignore the drive which is not what we really want here. By default the number of workers at NumCPUs/2 (50%) you can reduce it to '1' or something

```
_MINIO_HEAL_WORKERS=1
```

**Comment by beck-8** [comment]
@ravindk89 
I'd be happy to provide more data. 
I can reproduce this problem more easily in a larger cluster, do you mind? Because last time another colleague said my cluster was too big and ignored my problem.

**Comment by beck-8** [comment]
I have tried using mc admin config set heal to lower the healing, but it did not improve

**Comment by beck-8** [comment]
I will reproduce the issue again with the latest version at daytime UTC+8 and provide you with more information.

**Comment by harshavardhana** [comment]
> @ravindk89
> I'd be happy to provide more data.
> I can reproduce this problem more easily in a larger cluster, do you mind? Because last time another colleague said my cluster was too big and ignored my problem.

@beck-8, any performance issues require a subscription. We do not provide that as a free service for community users.

We have customers healing their drives on a daily basis; what you have is a setup problem which is 

- slow drive mostly HDDs with small IOPs
- slow network

**Comment by beck-8** [comment]
I have said it more than once, this is not a performance issue, it is a code bug.
My hard drive is an enterprise-class HDD, with normal standard performance; the network is 25Gbps*2
You didn't understand the problem I was talking about at all. During the repair process, reading data that has not been repaired will be stuck. At this time, I stopped the broken machine directly and it recovered instantly, so I concluded that there was a problem with the reading strategy or there was a lock that should not be generated.
